src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenMscorlib.cs

```diff
@@ -252,6 +252,44 @@ static int M(decimal d)
             var verifier = CompileAndVerify(compilation2);
         }
 
+        public void NoTypedRef()
+        {
+            var source1 =
+@"namespace System
+{
+    public class Object { }
+    public struct Void { }
+    public class ValueType { }
+    public struct Int32 { }
+    public struct Decimal { }
+}";
+            var compilation1 = CreateCompilation(source1, assemblyName: GetUniqueName());
+            var reference1 = MetadataReference.CreateFromStream(compilation1.EmitToStream());
+            var source2 =
+@"    
+public class C1
+{
+    public static T Read<T>()
+    {
+        T result = default(T);
+        var refresult = __makeref(result);
+
+        // ... method body
+
+        return result;
+    }
+}
+";
+            var compilation2 = CreateCompilation(source2, new[] { reference1 });
+            // Should report "CS0656: Missing compiler required member 'System.Decimal.op_Explicit_ToInt32'".
+            // Instead, we report no errors and assert during emit.
```

Comment:
I don't quite understand this comment.  It says we report no errors but then just below we verify an error is produced.